### PR TITLE
GitHub login oauth

### PR DESCRIPTION
**Pull request by erich**
https://trello.com/c/PsZKDjPo/6-as-an-user-with-github-profile-i-login-via-github-oauth-list-of-allowed-github-emails

### CODE REVIEW

**Comment by Math3v** [comment]
Hello, @erich! This is your first Pull Request that will be reviewed by [Ebert](https://ebertapp.io/), an automatic Code Review service. It will leave comments on this diff with potential issues and style violations found in the code as you push new commits. You can also see all the issues found on this Pull Request on its [review page](https://ebertapp.io/github/PrimeHammer/reddex/pulls/12). Please check our [documentation](https://docs.ebertapp.io) for more information.

File: lib/reddex_web/controllers/auth_controller.ex [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def callback(%{assigns: %{ueberauth_failure: _fails}} = conn, _params) do
+    conn
+    |> put_flash(:error, "Failed to authenticate.")
+    |> redirect(to: "/")
+  end
+
+  def callback(%{assigns: %{ueberauth_auth: auth}} = conn, _params) do
+    # TODO: move? Where? ENV variable?
```
Comment (Math3v) [inline lib/reddex_web/controllers/auth_controller.ex]: TODO found

File: lib/reddex_web/router.ex [hunk 2]
```diff
@@ -19,6 +19,15 @@ defmodule ReddexWeb.Router do
     pipe_through(:browser)
 
     get("/", PageController, :index)
+    resources("/users", UserController, only: [:index])
+  end
+
+  # TODO: move me
```
Comment (Math3v) [inline lib/reddex_web/router.ex]: TODO found

**Comment by Math3v** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 2 possible new issues (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/PrimeHammer/reddex/pulls/12.

**Comment by Math3v** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 2 possible new issues (including those that may have been commented here).

But beware that this branch is **[13 commits behind the `PrimeHammer:master` branch](https://github.com/PrimeHammer/reddex/compare/PrimeHammer:github-login-oauth...PrimeHammer:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/PrimeHammer/reddex/pulls/12.

File: lib/reddex_web/controllers/auth_controller.ex (hunk 1 above)
Comment (erich) [reply lib/reddex_web/controllers/auth_controller.ex]: Will fix later

File: lib/reddex_web/router.ex (hunk 2 above)
Comment (erich) [reply lib/reddex_web/router.ex]: Will fix later

**Comment by Math3v** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 2 possible new issues (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/PrimeHammer/reddex/pulls/12.

**Pull request merged**